simba/schedule.py

```diff
@@ -604,6 +628,43 @@ def generate_scenario(self, args):
                             ext_load["csv_file"] = str(ext_load_path)
                             events["fixed_load"][gc_name + " ext. load"] = ext_load
 
+                        # temporary lowering of grid connector max power during peak load windows
+                        if time_windows is not None:
+                            grid_operator = station.get('grid_operator')
+                            if grid_operator is None:
+                                warnings.warn('Peak Load Windows given, '
+                                              f'but {gc_name} has no grid operator')
+                            # get relevant peak load windows
+                            windows = time_windows.get(grid_operator)
+                            if grid_operator is not None and windows is None:
+                                warnings.warn(f'Grid operator {grid_operator} '
+                                              'not in time windows')
```

Comment:
the corresponding grid operator must be declared in the time windows file. Otherwise, it leads to unexpected behavior (time windows file given, but has no effect). Warning still present.